app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -105,7 +120,7 @@ def select
           best_tenant = nil
           best_flavor = nil
           instantiation_cost = Float::INFINITY
-          tmpls.each do |tmpl|
+          @tmpls.each do |tmpl|
```

Comment:
1 access method should be used in the whole code. So only `@tmpl` and remove `tmpl` getter or use `tmpl` getter. Using both is misleading.